dashboards/clicks_and_impressions.py

```diff
@@ -0,0 +1,92 @@
+import streamlit as st
+import matplotlib.pyplot as plt
+
+from utils.duckdb import get_dbcur
+
+st.title("Clicks & Impressions Dashboard")
+
+cur = get_dbcur()
+
+query = """
+WITH ranked_events AS (
+    SELECT 
+        ed.website_event_id,
+        es.id AS mentor_id,
+        ANY_VALUE(es.name) AS mentor_name,
+        SUM(CASE WHEN we.event_name = 'Click' THEN 1 ELSE 0 END) AS click_count,
+        SUM(CASE WHEN we.event_name = 'Impression' THEN 1 ELSE 0 END) AS impression_count
+    FROM 
+        umamidb.event_data ed
+    JOIN 
+        umamidb.website_event we
+        ON ed.website_event_id = we.event_id
+    JOIN 
+        memory.elasticsearch es
+        ON es.id = ed.string_value
+    WHERE 
+        (we.event_name = 'Click' OR we.event_name = 'Impression')
+        AND we.url_path LIKE '/mentors%'
+        AND we.referrer_path LIKE '/mentors%'
+    GROUP BY 
+        ed.website_event_id, 
+        es.id
+)
+
```

Comment:
From my understanding, each row of `ranked_events` should correspond to a single row in `event_data` with `ed.data_key = 'id'`, as well a single row in `website_event` and `elasticsearch` each since they have primary keys `event_id` and `id` respectively. (Technically, `elasticsearch` doesn't have a primary key since it's from JSON data, but `id` should be unique...)

If this is correct, could we run a check that `click_count + impression_count = 1` (or XOR) for each row?